SDALib-CPP/SDAReplay/ReplayMain.cpp

```diff
@@ -1,11 +1,41 @@
 #include "ReplayDriver.h"
+#include <boost/program_options.hpp>
 
-int main()
+using namespace boost::program_options;
+
+int main(int p_argc, char **p_path)
 {
-    std::string replayFileName("ReplayData/decisions.txt");
+    variables_map vm;                                // variables_map that keeps track of the amount of arguments it gets fed through command lines.
+    options_description argOptions{"CMDArguments"};  // initialize the command-line options.
+
+    // determines all arguments options (name), the value it should have (s) and a description of what it is (description)
+    argOptions.add_options()("bbfile", value<std::string>(), "BlackBoxReplayFile");
+
+    // tries to add all command line arguments, if an argument does not exists as an option in "argOptions": print out an error.
+    try
+    {
+        store(parse_command_line(p_argc, p_path, argOptions), vm);
+    }
+    catch (const error &er)
+    {
+        std::cerr << "argument is invalid, try: \"--bbfile\" instead" << std::endl;
+        return 0;
+    }
+
+    std::string replayFileName;  // initialize the string
+    if (vm.count("bbfile"))
+    {
+        replayFileName = vm["bbfile"].as<std::string>();
+    }
+
+    if (replayFileName.empty())  // if you don't give the program a filename, print the following error.
+    {
+        std::cerr << "no replay file found, please use command \"--bbfile <filename>\"" << std::endl;
+        return 0;
```

Comment:
this should return an error code.
```suggestion
        return 2;
```